src/track/beats.h

```diff
@@ -1,47 +1,77 @@
-#ifndef BEATS_H
-#define BEATS_H
+#pragma once
 
-#include <QObject>
-#include <QString>
-#include <QList>
 #include <QByteArray>
+#include <QList>
+#include <QObject>
 #include <QSharedPointer>
+#include <QString>
 
+#include "proto/beats.pb.h"
+#include "track/bpm.h"
+#include "util/logger.h"
 #include "util/memory.h"
 #include "util/types.h"
 
-namespace {
-    double kMaxBpm = 500;
-}
-
 namespace mixxx {
 
 class Beats;
-typedef QSharedPointer<Beats> BeatsPointer;
+using BeatsPointer = std::shared_ptr<Beats>;
+using BeatList = QList<track::io::Beat>;
+using SamplePos = double;
+using FrameNum = double;
+} // namespace mixxx
 
-class BeatIterator {
-  public:
-    virtual ~BeatIterator() {}
-    virtual bool hasNext() const = 0;
-    virtual double next() = 0;
-};
+#define BEAT_MAP_VERSION "BeatMap-1.0"
+// TODO(JVC) This two are deprecated
+#define BEAT_GRID_1_VERSION "BeatGrid-1.0"
+#define BEAT_GRID_2_VERSION "BeatGrid-2.0"
 
-// Beats is a pure abstract base class for BPM and beat management classes. It
-// provides a specification of all methods a beat-manager class must provide, as
-// well as a capability model for representing optional features.
+#include "track/beatiterator.h"
+#include "track/timesignature.h"
+#include "track/track.h"
+
+namespace {
+}
+
+namespace mixxx {
+/// Beats is a class for BPM and beat management classes.
+/// It stores beats information including beats position, down beats position,
+/// phrase beat position and changes in tempo.
+// TODO(JVC) To make it final
 class Beats : public QObject {
     Q_OBJECT
   public:
-    Beats() { }
-    virtual ~Beats() { }
+    /// Construct a BeatMap. iSampleRate may be provided if a more accurate
```

Comment:
```suggestion
    /// iSampleRate may be provided if a more accurate
```
This is a constructor, of course it constructs the class.